oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -430,6 +445,20 @@ public Builder jwkSetUri(String jwkSetUri) {
 			return this;
 		}
 
+		/**
+		 * Sets the metadata describing the provider's configuration.
+		 *
+		 * @since 5.1
+		 * @param configurationMetadata the metadata describing the provider's configuration
+		 * @return the {@link Builder}
+		 */
+		public Builder providerConfigurationMetadata(Map<String, Object> configurationMetadata) {
+			if (!CollectionUtils.isEmpty(configurationMetadata)) {
```

Comment:
This is not actually setting the value (as the Javadoc states) if it is empty. We should avoid setting the value to null, but we should not avoid setting it to an empty collection if the input is empty.

Reply:
I applied the update @rwinch